pep-0458.txt

```diff
@@ -188,9 +186,6 @@ The threat model assumes the following:
 
 * Offline keys are safe and securely stored.
 
-* Attackers can compromise at least one of PyPI's trusted keys stored online,
```

Comment:
Sorry, why was this removed?

Reply:
This is not part of the minimum security model. An attacker that compromises the online key(s) in PEP 458 can install malicious updates (although recovery is possible).

Reply:
Sorry, I'm not sure I agree with this. OFC attackers who compromise the repository _can_ access these online keys, right? That is why we have the security analysis tables in the PEP.

Reply:
Yes, this is possible for attackers to do, but if they do, they successfully attack the system. Per the end of this section: "An attacker is considered successful if it can cause a client to install (or leave installed) something other than the most up-to-date version of a software distribution file."

Maybe a better clarification would be to leave this bullet point and clarify later in the section that the goal is recovery from an attack, not just prevention.

Reply:
I was thinking we would keep the bullet point, but change "can" to "*cannot*". And then note after the bullet points that PEP 480 changes that assumption in order to address compromises of PyPI itself.

Reply:
I like @ncoghlan's solution!